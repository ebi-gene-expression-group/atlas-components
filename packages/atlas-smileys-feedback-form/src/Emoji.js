import React from 'react'
import PropTypes from 'prop-types'

const emojiData ={
  smiley: `😃`,
  disappointed: `😞`,
  pensive: `😔`,
  neutral_face: `😐`,
  grin: `😁`,
  satisfied: `😆`
}

const Emoji = ({emoji, onLeave, onOver, onClick}) =>{
  return   <span
    onClick={(e) => onClick(e)}
    onMouseEnter={(e) => onOver(e)}
    onMouseLeave={(e) => onLeave(e)}
    id={emoji}
    style={{fontSize: `40px`}}
  >
    {emojiData[emoji]}
  </span>
}

Emoji.propTypes={
  emoji: PropTypes.string.isRequired,
  onLeave: PropTypes.func.isRequired,
  onOver: PropTypes.func.isRequired,
  onClick: PropTypes.func.isRequired
}

export default Emoji