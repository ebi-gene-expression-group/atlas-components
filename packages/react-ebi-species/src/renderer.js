"use strict";

//*------------------------------------------------------------------*

var React = require('react');
var ReactDOM = require('react-dom');

//*------------------------------------------------------------------*

var Icon = require('./SpeciesIcon.jsx');

//*------------------------------------------------------------------*

module.exports = function(mountNode,species,colourOverride, colourPerKingdom) {
    ReactDOM.render(
        React.createElement(Icon,{
          species: species,
          colourOverride: colourOverride,
          colourPerKingdom: colourPerKingdom
        }), mountNode
    );
};
