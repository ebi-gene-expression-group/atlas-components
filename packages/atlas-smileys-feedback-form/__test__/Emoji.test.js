import React from 'react'
import renderer from 'react-test-renderer'
import Enzyme from 'enzyme'
import {shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import Emoji from '../src/Emoji'

Enzyme.configure({ adapter: new Adapter() })

describe(`Emojione`, () => {

  const smileyId = [`disappointed`, `slightly_frowning_face`, `neutral_face`, `grin`, `satisfied`]
  const randoemFace = smileyId[Math.round(Math.random(0,smileyId.length))]
  const props = {
    emoji: randoemFace,
    onLeave: ()=>{},
    onOver: ()=>{},
    onClick: ()=>{},
  }

  test(`should render 2 span components and id equals to emoji id`, () => {
    const wrapper = shallow(<Emoji {...props} />)
    expect(wrapper.find(`span`)).toHaveLength(1)
    expect(wrapper.find(`#${randoemFace}`)).toHaveLength(1)
  })

  test(`matches snapshot`, () => {
    const tree = renderer.create(<Emoji  {...props} emoji={`grin`}/>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})