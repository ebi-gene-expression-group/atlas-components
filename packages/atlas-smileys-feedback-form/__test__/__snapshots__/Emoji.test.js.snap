// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emojione matches snapshot 1`] = `
<span
  id="grin"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "fontSize": "40px",
    }
  }
>
  😁
</span>
`;
